### ISSUE TITLE
Mount failed on Windows nodes and causing blue screens

### ISSUE THREAD

**Issue by twecker137** [kubernetes-csi/csi-driver-smb#211]
**What happened**:

SMB mounts cannot be mounted on Windows systems. In addition, a blue screen occurs in 50% of the attempts.

**What you expected to happen**:

SMB mounts can be mounted from Windows systems.

**How to reproduce it**:

We use an on-premise Rancher Kubernetes cluster running Windows as well as Linux nodes. We have installed the Helm Chart csi-driver-smb-v0.6.0 in the kube-system namespace, with Windows integration enabled.

In addition, we have created a Storage class and PV/PVCs according to the following templates:

*StorageClass*
```yaml
---
apiVersion: storage.k8s.io/v1
kind: StorageClass
metadata:
  name: mount-readwritemany
parameters:
  createSubDir: "false"
provisioner: smb.csi.k8s.io
reclaimPolicy: Retain
volumeBindingMode: Immediate
```

*PV*
```yaml
---
apiVersion: v1
kind: PersistentVolume
metadata:
  name: smb-share-test
spec:
  storageClassName: mount-readwritemany
  capacity:
    storage: 20Gi 
  accessModes:
    - ReadWriteMany
  persistentVolumeReclaimPolicy: Retain
  csi:
    driver: smb.csi.k8s.io
    readOnly: false                  
    volumeHandle: share-smb1  # make sure it's a unique id in the cluster
    volumeAttributes:
      source: "//10.99.99.1/share" 
    nodeStageSecretRef:
      name: smbcreds-win # secret with Windows credentials
      namespace: default
```
*PVC*
```yaml
---
kind: PersistentVolumeClaim
apiVersion: v1
metadata:
  name: pvc-smb-share-test
spec:
  storageClassName: mount-readwritemany  
  accessModes:
    - ReadWriteMany
  resources:
    requests:
      storage: 20Gi  
  volumeName: smb-share-test
```

**Anything else we need to know?**:

Manually mounting the share from one of the Windows nodes is successfully. All windows nodes within the cluster are affected.

There are two error messages that occur in the context:

```utils.go:111] GRPC error: rpc error: code = Internal desc = volume(share-smb1) mount "//10.99.99.1/share" on "c:\\var\\lib\\kubelet\\plugins\\kubernetes.io\\csi\\pv\\smb-share-test\\globalmount" failed with smb mapping failed with error: rpc error: code = Unknown desc = NewSmbGlobalMapping failed. output: "New-SmbGlobalMapping : Generic failure \r\nAt line:1 char:190\r\n+ ... ser, $PWord;New-SmbGlobalMapping -RemotePath $Env:smbremotepath -Cred ...\r\n+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\r\n    + CategoryInfo          : NotSpecified: (MSFT_SmbGlobalMapping:ROOT/Microsoft/...mbGlobalMapping) [New-SmbGlobalMa \r\n   pping], CimException\r\n    + FullyQualifiedErrorId : HRESULT 0x80041001,New-SmbGlobalMapping\r\n \r\n", err: exit status 1```

This error message occurs often together with a bluescreen:

```MountVolume.MountDevice failed for volume "smb-share-test" : rpc error: code = Internal desc = volume(share-smb1) mount "//10.99.99.99/share" on "c:\\var\\lib\\kubelet\\plugins\\kubernetes.io\\csi\\pv\\smb-share-test\\globalmount" failed with smb mapping failed with error: rpc error: code = Unknown desc = NewSmbGlobalMapping failed. output: "New-SmbGlobalMapping : Multiple connections to a server or shared resource by the same user, using more than one user \r\nname, are not allowed. Disconnect all previous connections to the server or shared resource and try again. \r\nAt line:1 char:190\r\n+ ... ser, $PWord;New-SmbGlobalMapping -RemotePath $Env:smbremotepath -Cred ...\r\n+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\r\n + CategoryInfo : NotSpecified: (MSFT_SmbGlobalMapping:ROOT/Microsoft/...mbGlobalMapping) [New-SmbGlobalMa \r\n pping], CimException\r\n + FullyQualifiedErrorId : Windows System Error 1219,New-SmbGlobalMapping\r\n \r\n", err: exit status 1```

The global mappings are created sucessfully in any case:

```powershell
Get-SmbGlobalMapping

Status Local Path Remote Path
------ ---------- -----------
OK                \\10.99.99.1\share
```

**Environment**:

- CSI Driver version: v0.6.0
- Kubernetes version: v1.18.10
- OS: Windows Server 2019, Version 1809, OS Build 17763.1339

**Comment by andyzhangx** [comment]
what version of csi-proxy are you using? We suggest installing v0.2.2:
https://github.com/kubernetes-csi/csi-driver-smb/blob/9b6052080db256effcd121e26e5acb6263e03f19/.github/workflows/windows.yaml#L27

And interesting thing is why there is blue screen after `New-SmbGlobalMapping` operation?

**Comment by twecker137** [comment]
When analyzing the problem, we were able to identify two sources of error:
First, the SMB share could not be mounted because DFS was configured on the target SMB server and `New-SmbGlobalMapping`  simply does not support it.
Second, we rely on Rancher RKE for the Kubernetes cluster. In doing so, we have seen in the logs that the mounts on the Windows servers appear to be attempted via Unix sockets, which is likely the cause of the blue screens. We will therefore close this issue and probably open one for Rancher if we are able to reproduce the bug again.
As a solution, we have redeployed our file server to a solution without DFS, and are using a flexVolume driver instead for now.
